### ISSUE TITLE
Handle nullable y in sklearn classifiers

### ISSUE THREAD

**Issue by tamargrey** [alteryx/evalml#3910]
The following estimators will raise `ValueError: Unknown label type: 'unknown'` if a `y` with a nullable logical type is passed in:
- ElasticNetClassifier
- DecisionTreeClassifier
- ExtraTreesClassifier
- KNeighborsClassifier
- LogisticRegressionClassifier
- RandomForestClassifier
- SVMClassifier

These are all of the classifiers that come from sklearn

The following test will raise this error for any of the above classifiers. Note setting the logical type to `Integer` or `Boolean` does not raise this error.

**Classifiers**
```python
def test_nullable_y(X_y_binary):
    import woodwork as ww
    X, y = X_y_binary

    # Using either of these lines results in the same error
    y = ww.init_series(y, logical_type="IntegerNullable")
    # y = ww.init_series(y, logical_type="BooleanNullable")

    # Using any of the estimators results in the error 
    clf = ElasticNetClassifier()

    clf.fit(X, y)
```

I suspect that currently, we avoid running into this problem in AutoMLSearch because of our `ReplaceNullableTypes` transformer. To support nullable types across EvalML, we should move nullable-specific handling to the individual components that need it. Since this seems to be a common error between all of the sklearn classifiers, I'd like to see if theres a way for us to introduce nullability handling without having to have repeated logic in all of these components.

**Comment by tamargrey** [comment]
There are a many tests that, when nullable types are forcibly used (as they are in [this commit](https://github.com/alteryx/evalml/pull/3895/commits/cb4e1ee866d40dd5cebe274904cfd50880d1193b)) will fail with this error. I believe the vast majority are because of the estimators listed above, but we should confirm that, once a fix is put in place for the components above, all of the appearances of this error disappear when used with nullable types.

One example of a component that seems to have this error but not be a part of the above components is the `Oversampler`, as `test_no_oversample[pd-numeric]` failed at `new_X, new_y = oversampler.fit_transform(X, y)`.

Below is list of tests (not comprehensive!) that failed that were not obviously connected with one of the above components. As mentioned, it's likely that the pipelines they're using use these components, but we should confirm that.

- test_train_pipeline_trains_and_tunes_threshold_ts
- test_submit_training_job_single[threads]
- test_submit_training_jobs_multiple[threads]
- test_submit_scoring_job_single[processes]
- test_cfengine_sends_woodwork_schema[processes]
- test_submit_training_job_single
- test_submit_scoring_job_single
- test_fit_predict_binary
- test_feature_importance
- test_fit_predict_multi
- test_component_fit - after make_rf_feature_selectors
- test_stacked_ensemble_init_with_final_estimator
- test_no_oversample[pd-numeric] 
- test_find_confusion_matrix_pipeline_threshold[0.5-20]
- test_partial_dependence_problem_types[multiclass-np]
- test_explain_predictions_oversampler[LightGBMClassifier-lime]
- test_transform_all_but_final

**Comment by tamargrey** [comment]
Also seen with:
- StackedEnsembleClassifier
- LightGBMClassifier

**Comment by tamargrey** [comment]
a decent number of these may be fixed by upgrading to sklearn 1.2.1

**Comment by tamargrey** [comment]
Closing with https://github.com/alteryx/evalml/pull/3983 being merged and released. Any remaining incompatibilities will have their own tickets.
